base/shell/explorer/startctxmnu.cpp

```diff
@@ -168,49 +174,48 @@ class CStartMenuBtnCtxMenu :
                     hRet = psfDesktop->BindToObject(pidlStart, NULL, IID_PPV_ARG(IShellFolder, &m_Folder));
                     if (SUCCEEDED(hRet))
                     {
-                        hRet = CreateContextMenuFromShellFolderPidl(hPopup);
-                        m_idCmdCmLast = (SUCCEEDED(hRet)) ? HRESULT_CODE(hRet) : ID_SHELL_CMD_LAST;
-                        AddStartContextMenuItems(hPopup);
+                        hRet = CreateContextMenuFromShellFolderPidl(hPopup, idInnerFirst, idCmdLast);
                     }
                 }
             }
 
             ILFree(pidlStart);
         }
-
-        return S_OK;
+        if (idCmdLast - idCmdFirst >= ID_SHELL_CMD_LAST - ID_SHELL_CMD_FIRST)
+        {
+            AddStartContextMenuItems(hPopup);
```

Comment:
Can this code block be run even if the CreateContextMenuFromShellFolderPidl() call from above failed (or wasn't executed because either pidlStart was NULL or some of the if-tests failed) ?

Reply:
Yes, you would then end up with a menu that just has Properties, Separator, Open/Explore All Users.